src/Databases/DatabaseDictionary.cpp

```diff
@@ -93,8 +93,7 @@ ASTPtr DatabaseDictionary::getCreateTableQueryImpl(const Context & context,
         WriteBufferFromString buffer(query);
 
         const auto & dictionaries = context.getExternalDictionariesLoader();
-        auto dictionary = throw_on_error ? dictionaries.getDictionary(table_name)
-                                         : dictionaries.tryGetDictionary(table_name);
+        auto dictionary = dictionaries.tryGetDictionary(table_name, false /*load*/);
         if (!dictionary)
             return {};
```

Comment:
Nope, the structure is known even without filling the dictionary, and also this is covered by the [test](https://github.com/ClickHouse/ClickHouse/pull/10164/files#diff-f58da075bbb4b6f337032c9176ea5baeR20) (and I didn't use `FORMAT Null` to double verify this), which passes (but I just rebased, due to missing forward declaration for one error code)

Reply:
>and also this is covered by the test

Ok, this is not correct, since this is another database engine, I just checked this case and it does not works, plus one cannot return empty `ASTPtr` from the `getCreateTableQueryImpl` since `InterpreterShowCreateTable` cannot handle this, will be fixed

Reply:
That became possible after `throw_on_error` check had been removed, so I just reverted that hunk, and make it loads the dictionary